### PR TITLE
MNTOR-2329/switch from mozlog to winston

### PR DESCRIPTION
**Pull request by rhelmer**
# References: 
Jira: MNTOR-2329

# Description

Remove `mozlog` and switch to `winston`.

# How to test

Outside of stage/prod, this should log to the console only. In GCP, it will use cloud logging and parse the payload into `jsonPayload` correctly (which can be used from stackdriver and BigQuery).

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.

### CODE REVIEW

File: src/app/api/utils/auth.ts [hunk 1]
```diff
@@ -172,10 +168,10 @@ export const authOptions: AuthOptions = {
   },
   events: {
     signIn(message) {
-      log.debug("fxa-confirmed-profile-data", message.user);
+      logger.debug("fxa-confirmed-profile-data", message.user);
```
Comment (mansaj) [inline src/app/api/utils/auth.ts:171]: PII check?

Comment (mansaj) [reply src/app/api/utils/auth.ts:171]: oh since it's debug, prob won't log to gcp

File: src/app/functions/server/onerep.ts [hunk 2]
```diff
@@ -154,7 +153,7 @@ export async function getProfile(
     method: "GET",
   });
   if (!response.ok) {
-    log.info(
+    logger.info(
```
Comment (mansaj) [inline src/app/functions/server/onerep.ts:156]: We might want to change this to error or warning?

File: src/app/functions/server/onerep.ts [hunk 3]
```diff
@@ -127,7 +126,7 @@ export async function createProfile(
     body: JSON.stringify(requestBody),
   });
   if (!response.ok) {
-    log.info(
+    logger.info(
```
Comment (mansaj) [inline src/app/functions/server/onerep.ts:129]: We might want to change this to error or warning?

**Review by mansaj — APPROVED** [review]

File: src/app/api/utils/auth.ts (hunk 1 above)
Comment (rhelmer) [reply src/app/api/utils/auth.ts:171]: Thanks... I think we're capturing debug, I'll make it log something here that we can use for diagnostics but not the whole user object.

File: src/app/functions/server/onerep.ts (hunk 3 above)
Comment (rhelmer) [reply src/app/functions/server/onerep.ts:129]: Hm yeah these are definitely errors, thanks

**Comment by Vinnl** [comment]
Woah, great cleanup Rob, thanks for that!

**Pull request merged**